### PR TITLE
Add a trigger_on_create kwarg

### PR DESCRIPTION
**Pull request by craigds**
At present the `post_save` and `pre_save` don't trigger when you call `MyModel.objects.create(...)`

It makes some sense that they should. This PR adds that if you specify a `trigger_on_create=True` arg when connecting the signal receiver.

# to do

- [ ] Add a test for trigger_during_create=True
- [ ] Decide on appropriate value for `changed_fields` in this situation

## open question on changed_fields value

I'm not 100% sure what to populate changed_fields with:

1. no fields (`changed_fields={}`). This is the easy thing - no fields changed between init and save.
2. all fields specified during signal connection (the instance is being created, so all fields have changed).

If (2) then what should `old_value` be in the `changed_fields` dict?

1. `None`
2. `field.default or None`
3. the same as `new_value`

### CODE REVIEW

File: fieldsignals/signals.py [hunk 1]
```diff
@@ -91,20 +104,28 @@ def connect_source_signals(self, sender):
...
             changed_fields = self.get_and_update_changed_fields(
-                receiver, instance, fields
+                receiver,
+                instance,
+                fields,
+                trigger_during_create=(created and trigger_during_create),
             )
-            if changed_fields:
+
+            if changed_fields or (created and trigger_during_create):
```
Comment (benpearman) [inline fieldsignals/signals.py:122]: Adding `(created and trigger_during_create)` here - is that redundant?

`changed_fields` will be non-empty, as `trigger_during_create` is supplied to  `get_and_update_changed_fields` above

Comment (benpearman) [reply fieldsignals/signals.py:122]: Ah - but not in the case that there are only deferred fields

**Comment by benpearman** [comment]
For (2) - pretty arbitrary but:
`changed_fields` will only be populated if they have manually overridden `trigger_during_create` as `True`

In that situation, what information is most useful for them? Probably option 2.

I think option 1 and 3 are more easily implementable by the user in their own code, compared to option 2.  
And not option 3, since IMO it more readily defies the variable name `old_value`.

**Review by benpearman — APPROVED** [review]

**Comment by craigds** [comment]
I'm re-thinking this and I'm not really sure it adds much.
It has to be opt-in I think; it can't be the default behaviour, because it's not _obviously_ what you want as a user.

If you do want to enable this behaviour, it's not very hard to explicitly enable it by just adding a post_save handler as well as a post_save_changed handler:

```python
@receiver(post_save, sender=MyModel)
@receiver(post_save_changed, sender=MyModel, fields=["field1"])
def on_mymodel_created_or_changed(sender, instance, *, changed_fields=None, created, **kwargs):
    if not (created or changed_fields):
        return
    ...
```

That does the same thing as this PR and it might be more explicit.

Maybe a brief note in the README would be helpful to state clearly that fieldsignals don't trigger during create

**Comment by craigds** [comment]
In fddbcad1c83121b95ad75e8a60c98cbdf8454778 I've added a note to the README about this

**Pull request closed without merging**